lib/bundler/shared_helpers.rb

```diff
@@ -210,14 +210,11 @@ def pretty_dependency(dep, print_source = false)
 
     def md5_available?
       return @md5_available if defined?(@md5_available)
-      @md5_available = begin
-        require "openssl"
-        OpenSSL::Digest::MD5.digest("")
-        true
-      rescue LoadError
-        true
-      rescue OpenSSL::Digest::DigestError
+      require "openssl"
```

Comment:
We should still support old versions of OpenSSL that does not define `OpenSSL.fips_mode`

```suggestion
      @md5_available = begin
        require "openssl"
        if defined?(OpenSSL::OPENSSL_FIPS) && OpenSSL.respond_to?(:fips_mode)
          OpenSSL.fips_mode
        else
          OpenSSL::Digest::MD5.digest("")
          true
        end
      rescue LoadError
        true
      rescue OpenSSL::Digest::DigestError
        false
      end
```

Reply:
This makes sense, I will update it, glad you thought of backward compatibility.